src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_time_stm32f4_nucleo_wifi_sntp.h

```diff
@@ -0,0 +1,46 @@
+#ifndef __XI_BSP_TIME_STM32F4_NUCLEO_WIFI_SNTP_H__
+#define __XI_BSP_TIME_STM32F4_NUCLEO_WIFI_SNTP_H__
+
+#include <stdint.h>
+
+#define SNTP_SERVER "3.pool.ntp.org"
+#define SNTP_PORT   123
+
+typedef int32_t posix_time_t;
+typedef enum {
+    SNTP_INTERNAL_ERROR  = -6,
+    SNTP_NOT_AVAILABLE   = -5,
+    SNTP_PARSER_ERROR    = -4,
+    SNTP_REQUEST_FAILURE = -3,
+    SNTP_SOCKET_ERROR    = -2,
+    SNTP_TIMEOUT         = -1,
+    SNTP_SUCCESS         = 0,
+} sntp_status_t;
+
+uint8_t* sntp_sock_id_ptr;
```

Comment:
This shouldn't be a pointer, or the socket data callback will crash. I'm fixing it, will push in a minute